daemon/emer-daemon.c

```diff
@@ -1123,6 +1142,379 @@ set_max_bytes_buffered (EmerDaemon *self,
+                                   NULL);
+
+  tallies_folder = g_file_new_for_path (tallies_path);
+
+  enumerator = g_file_enumerate_children (tallies_folder,
+                                          "standard::*",
+                                          G_FILE_QUERY_INFO_NONE,
+                                          NULL,
+                                          &error);
+
+  if (error)
+    {
+      if (!g_error_matches (error, G_IO_ERROR, G_IO_ERROR_NOT_FOUND))
+        g_warning ("Error listing tallies from %s: %s",
+                   tallies_path, error->message);
+      return;
+    }
+
+  while (TRUE)
+    {
+      g_autoptr(GVariant) variant = NULL;
+      g_autoptr(GVariant) cache_key = NULL;
+      g_autoptr(GVariant) event_id = NULL;
+      g_autoptr(GVariant) aggregate_key = NULL;
+      g_autoptr(GVariant) payload = NULL;
+      g_autoptr(GError) local_error = NULL;
+      g_autofree gchar *contents = NULL;
+      g_autofree gchar *path = NULL;
+      GFileInfo *info;
+      GFile *file;
+      gboolean complete;
+      guint32 counter;
+      uid_t unix_user_id;
+
+      g_file_enumerator_iterate (enumerator, &info, &file, NULL, &local_error);
+
+      if (local_error)
+        {
+          if (!g_error_matches (error, G_IO_ERROR, G_IO_ERROR_NOT_FOUND))
+            g_warning ("Error listing tallies from %s: %s",
+                       tallies_path, local_error->message);
+          return;
+        }
+
+      if (!file || !info)
+        break;
+
+      if (g_file_info_get_file_type (info) != G_FILE_TYPE_REGULAR)
+        continue;
+
+      path = g_file_get_path (file);
+      g_file_get_contents (path, &contents, NULL, &local_error);
+
+      if (local_error)
+        {
+          g_warning ("Error loading tally from file %s: %s",
+                     path, local_error->message);
+          continue;
+        }
+
+      variant = g_variant_parse (NULL, contents, NULL, NULL, &local_error);
+      if (local_error)
+        {
+          g_warning ("Error parsing tally from file %s: %s",
+                     path, local_error->message);
+          continue;
+        }
+
+      g_variant_get (variant,
+                     "(vvvuubmv)",
+                     &event_id,
+                     &aggregate_key,
+                     &cache_key,
+                     &unix_user_id,
+                     &counter,
+                     &complete,
+                     &payload);
+
+      if (!complete)
+        continue;
+
+      emer_daemon_record_aggregate_event (self, event_id,
+                                          date_string,
+                                          counter,
+                                          payload != NULL,
+                                          payload);
+    }
+}
+
+static gboolean
+clock_ticked_midnight_cb (gpointer user_data)
+{
+  EmerDaemon *self = user_data;
+
+  save_aggregate_timers_to_tally (self);
+  buffer_daily_aggregate_events_to_queue (self);
```

Comment:
We also need to "catch up" for any day(s) missed while the computer was switched off.

Suppose I switch off my computer at 6pm on August 31st, and turn it on again at 8am on August 1st. Then we will have missed both the midnight and end-of-month ticks.